PluginOracleNet/API/Discover/GetAllRelatedEntities.cs

```diff
@@ -0,0 +1,166 @@
+        private const string ColForeignTableName = "FOREIGN_TABLE_NAME";
+        private const string ColForeignColumn = "FOREIGN_COLUMN";
+        private const string ColRelationshipName = "SOURCE_CONSTRAINT";
+
+        private const string GetForeignKeysQuery = @"SELECT a.OWNER          SOURCE_TABLE_SCHEMA,
+      a.TABLE_NAME      SOURCE_TABLE_NAME,
+      a.COLUMN_NAME     SOURCE_COLUMN,
+      c_pk.OWNER        FOREIGN_TABLE_SCHEMA,
+      c_pk.TABLE_NAME   FOREIGN_TABLE_NAME,
+      a.COLUMN_NAME     FOREIGN_COLUMN,
+      c.CONSTRAINT_NAME SOURCE_CONSTRAINT
+FROM all_cons_columns a
+    INNER JOIN all_constraints c ON a.OWNER = c.OWNER
+        AND a.CONSTRAINT_NAME = c.CONSTRAINT_NAME
+    INNER JOIN all_constraints c_pk ON c.r_owner = c_pk.owner
+        AND c.R_CONSTRAINT_NAME = c_pk.CONSTRAINT_NAME
+WHERE c.CONSTRAINT_TYPE = 'R'
+    AND c.OWNER = '{0}'
+    AND a.TABLE_NAME = '{1}'
+ORDER BY c.TABLE_NAME, c_pk.TABLE_NAME, c.CONSTRAINT_NAME, a.POSITION";
+
+        public static async IAsyncEnumerable<RelatedEntity> GetAllRelatedEntities(
+            IConnectionFactory connFactory,
+            RepeatedField<Schema> schemas)
+        {
+            var conn = connFactory.GetConnection();
+
+            try
+            {
+                await conn.OpenAsync();
+
+                foreach (var schema in schemas)
+                {
+                    var schemaParts = DecomposeSafeName(schema.Id).TrimEscape();
+                    var schemaName = schemaParts.Schema;
+                    var tableName = schemaParts.Table;
+                    
+                    string query = string.Format(GetForeignKeysQuery, schemaName, tableName);
+
+                    var cmd = connFactory.GetCommand(query, conn);
+                    var reader = await cmd.ExecuteReaderAsync();
+
+                    var sourceResourceId = "";
+                    var sourceColumnsList = new List<string>();
+                    var foreignColumnsList = new List<string>();
+                    var lastRelationshipName = "";
+                    var lastForeignResourceId = "";
+
+                    var emitEntity = false;
+                    var carryColumns = false;
+                    var canRead = true;
+                    while (canRead || lastRelationshipName != "")
+                    {
+                        var relationshipName = "";
+                        var foreignResourceId = "";
+                        
+                        // delay evaluation so that the loop cycles at least once before finishing
+                        canRead = await reader.ReadAsync();
+                        if (canRead)
+                        {
+                            // read values from current row
+                            sourceResourceId = $"{Utility.Utility.GetSafeName(reader.GetValueById(ColSourceTableSchema).ToString()?.Trim(' '))}.{Utility.Utility.GetSafeName(reader.GetValueById(ColSourceTableName).ToString()?.Trim(' '))}";
```

Comment:
let's tidy this up in a similar way to what I did on DB2 
https://github.com/naveego/plugin-db2/pull/8/commits/3caf0aaa4287a86f151db9d8df6d098740dcf11a